### PR TITLE
Minor fixes and additions

### PR DESCRIPTION
**Pull request by pszal**
- replace ReadPublic() by DecodePublic() when creating and loading keys: the current implementation calls ReadPublic() even if public data is already accessible
- drop handle() from the ak interface: it is unnecessary
- add PrivPubBlobs() to attest.Key: to allow `attest`-agnostic  key marshaling

### CODE REVIEW

File: attest/application_key.go [hunk 1]
```diff
@@ -101,3 +102,8 @@ func (k *Key) Marshal() ([]byte, error) {
 func (k *Key) CertificationParameters() CertificationParameters {
 	return k.key.certificationParameters()
 }
+
+// PubPrivBlobs returns private and public blobs to be used by tpm2.Load().
+func (k *Key) PubPrivBlobs() ([]byte, []byte, error) {
```
Comment (ericchiang) [inline attest/application_key.go:107]: Want to just call this "Blobs()"? Or "Export()"?

**Review by ericchiang — APPROVED** [review]

File: attest/application_key.go [hunk 2]
```diff
@@ -101,3 +102,8 @@ func (k *Key) Marshal() ([]byte, error) {
 func (k *Key) CertificationParameters() CertificationParameters {
 	return k.key.certificationParameters()
 }
+
+// Blobs returns private and public blobs to be used by tpm2.Load().
+func (k *Key) Blobs() ([]byte, []byte, error) {
```
Comment (ericchiang) [inline attest/application_key.go:107]: Blah, sorry I didn't catch this. But please name the return values. 

func (k *Key) Blobs() (pub, priv []byte, err error)

File: attest/application_key.go (hunk 1 above)
Comment (pszal) [reply attest/application_key.go]: Done

File: attest/application_key.go (hunk 2 above)
Comment (pszal) [reply attest/application_key.go]: Done

**Pull request merged**